src/SprykerEco/Yves/Braintree/Theme/default/components/molecules/braintree-payment-form/braintree-payment-form.ts

```diff
@@ -0,0 +1,169 @@
+    }
+
+    protected onScriptLoad(): void {
+        if (!!this.braintreeClientToken) {
+            this.setupBraintree();
+        }
+    }
+
+    protected submitForm(nonce: string = ''): void {
+        this.initTokenValue(nonce);
+
+        this.form.submit();
+    }
+
+    protected createTokenField(): void {
+        const tokenInput = document.createElement('input');
+        tokenInput.setAttribute('type', 'hidden');
+        tokenInput.setAttribute('name', `${this.nonceInputName}`);
+        this.form.appendChild(tokenInput);
+    }
+
+    protected initTokenValue(nonce: string): void {
+        const nonceInputSelector = <HTMLInputElement>document.querySelector(`input[name='${this.nonceInputName}']`);
+        nonceInputSelector.value = nonce;
+    }
+
+    protected clearTokenValue(): void {
+        const nonceInputSelector = <HTMLInputElement>document.querySelector(`input[name='${this.nonceInputName}']`);
+
+        nonceInputSelector.value = '';
+    }
+
+    protected errorHandler(error: braintreeErrorConfig): void {
+        const paymentMethod = this.currentPaymentMethodValue;
+
+        this.removeErrorMessage();
+
+        if (paymentMethod === this.paymentMethodName) {
+            this.addErrorMessage(error.message);
+            return;
+        }
+
+        this.submitForm();
+    }
+
+    protected paymentMethodHandler(response: braintreeConfig): void {
+        const paymentMethod = this.currentPaymentMethodValue;
+        const isWrongMethodSelected = (paymentMethod === this.paymentMethodName && response.type !== this.paymentMethodTypeName);
+
+        this.removeErrorMessage();
+
+        if (isWrongMethodSelected) {
+            this.errorHandler({
+                message: this.braintreeErrorMessage
+            });
+            return;
+        }
+
+        this.submitForm(response.nonce);
+    }
+
+    protected loadBraintree(): void {
+        this.braintreeSetupSettings = {
+            onPaymentMethodReceived: this.paymentMethodHandler.bind(this),
+            onError: this.errorHandler.bind(this)
+        };
+    }
+
+    protected setupBraintree(): void {
+        this.loadBraintree();
+        braintree.setup(this.braintreeClientToken, this.integrationType, this.braintreeSetupSettings);
+    }
+
+    setCurrentPaymentMethod(): void {
+        this.paymentMethods.forEach((method: HTMLInputElement) => {
+            this.setCurrentPaymentMethodValue(method);
+        });
+    }
+
+    setCurrentPaymentMethodValue(method: HTMLInputElement): void {
+        if (method.checked) {
+            this.currentPaymentMethodValue = method.value;
+        }
+    }
+
+    addErrorMessage(message: string = ''): void {
+        const errorContainer = <HTMLElement>this.querySelector(`.${this.jsName}__error`);
+        this.showErrorMessage(errorContainer);
+
+        errorContainer.innerHTML = message;
+    }
+
+    showErrorMessage(errorContainer: HTMLElement): void {
+        errorContainer.classList.remove('is-hidden');
```

Comment:
move class `is-hidden` to component attribute

Reply:
fixed